assets/src/domain/eventEditor/services/apollo/mutations/useReorderEntities.ts

```diff
@@ -0,0 +1,127 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
+import { clone, reverse } from 'ramda';
+import { useDebouncedCallback } from 'use-debounce';
+
+import { EntityId } from '@appServices/apollo/types';
+import { getGuids } from '@appServices/predicates';
+import { Datetime, Ticket } from '../types';
+
+type Entity = Datetime | Ticket;
+
+interface ReorderEntitiesProps {
+	entityType: 'DATETIME' | 'TICKET';
+}
+
+interface CallbackArgs<E extends Entity> {
+	allEntities: Array<E>;
+	filteredEntities: Array<E>;
+	newIndex: number;
+	oldIndex: number;
+	updateEntityList: (updatedEntities: Array<E>) => void;
+}
+
+type SortCallback<E extends Entity> = (args: CallbackArgs<E>) => void;
+
+const REORDER_ENTITIES = gql`
+	mutation REORDER_ENTITIES($input: ReorderEspressoEntitiesInput!) {
+		reorderEspressoEntities(input: $input) {
+			ok
+		}
+	}
+`;
+
+interface ReorderEntities<E extends Entity> {
+	cancel: VoidFunction;
+	done: VoidFunction;
+	sortEntities: SortCallback<E>;
+}
+
+const useReorderEntities = <E extends Entity>({ entityType }: ReorderEntitiesProps): ReorderEntities<E> => {
+	const [allEntityGuids, setAllEntityGuids] = useState<Array<EntityId>>([]);
+	useEffect(() => {
+		if (allEntityGuids.length) {
+			done();
+		}
+	}, [allEntityGuids.join(':')]);
+
+	const [mutate, result] = useMutation(REORDER_ENTITIES);
+	useEffect(() => {
+		console.log(result);
+	}, [result]);
+
+	const [runMutation, cancelDebounce] = useDebouncedCallback((...args) => {
+		console.log('runMutation called');
+		mutate(...args);
+	}, 5000); // delay in MS
```

Comment:
I have set a delay of 5 seconds, it that OK?

Reply:
might be a bit long... we'll see